proto/transport.proto

```diff
@@ -0,0 +1,224 @@
+      NO_DIRECTION = 1;
+      UP = 2;
+      DOWN = 3;
+      IN = 4;
+      OUT = 5;
+      CLOCKWISE = 6;
+      ANTICLOCKWISE = 7;
+      FORWARD = 8;
+      BACKWARD = 9;
+      EAST = 10;
+      WEST = 11;
+      NORTH = 12;
+      SOUTH = 13;
+    }
+
+    enum DoorStatus {
+      DOOR_STATUS_UNSPECIFIED = 0;
+      CLOSED = 1;
+      OPEN = 2;
+      OPENING = 3;
+      CLOSING = 4;
+      // The door is safety locked.
+      SAFETY_LOCKED = 5;
+      // The door remains in a position that is not fully open or closed.
+      LIMITED_OPENED = 6;
+    }
+
+    enum Mode {
+      MODE_UNSPECIFIED = 0;
+      NORMAL = 1;
+      // The transport is in service or inspection control.
+      SERVICE_CONTROL = 2;
+      // The transport is under a firefighter transport or firefighting transport mode of control. This usually happens
+      // during a fire alarm when firemen are on the site.
+      FIREFIGHTER_CONTROL = 3;
+      OUT_OF_SERVICE = 4;
+      // The transport is being operated by emergency power and may be in limited operation.
+      EMERGENCY_POWER = 5;
+      // The transport will not respond to any instruction until the VIP has reached their destination.
+      VIP_CONTROL = 6;
+      EARTHQUAKE_OPERATION = 7;
+      FIRE_OPERATION = 8;
+      // The transport is being manually operated by an attendant.
+      ATTENDANT_CONTROL = 9;
+      // The transport is manually or automatically parked at a predefined floor and will not provide
+      // any further service. This usually happens in a low traffic condition for the purpose of
+      // energy saving.
+      PARKING = 10;
+      // The transport is returning to its home position, usually the lobby or ground floor.
+      HOMING = 11;
+      // Control of the transport is performed from a control panel in the control cabinet. The transport is not
+      // available for normal operation.
+      CABINET_RECALL = 12;
+      // The transport is under an occupant evacuation mode of control. This usually happens during a
+      // fire alarm when the transport is used for evacuation of occupants.
+      OCCUPANT_EVACUATION = 13;
+      // The transport is operational but is reserved for the transportation of goods only.
+      FREIGHT = 14;
+      FAILURE = 15;
+      REDUCED_SPEED = 16;
+      STORM_OPERATION = 17;
+      HIGH_WIND_OPERATION = 18;
+    }
+
+    enum Fault {
+      FAULT_UNSPECIFIED = 0;
+      CONTROLLER_FAULT = 1;
+      DRIVE_AND_MOTOR_FAULT = 2;
+      MECHANICAL_COMPONENT_FAULT = 3;
+      OVERSPEED_FAULT = 4;
+      POWER_SUPPLY_FAULT = 5;
+      SAFETY_DEVICE_FAULT = 6;
+      CONTROLLER_SUPPLY_FAULT = 7;
+      DRIVE_TEMPERATURE_EXCEEDED = 8;
+      // A comb plate safety switch is activated.
+      COMB_PLATE_FAULT = 9;
+      GENERAL_FAULT = 10;
+      DOOR_FAULT = 11;
+      LEVELLING_FAULT = 12;
+      SAFETY_CIRCUIT_BREAK_FAULT = 13;
+      FAIL_TO_START = 14;
+      ALARM_BUTTON = 15;
+      DOOR_NOT_CLOSING = 16;
+    }
```

Comment:
Seeing these faults and knowing we have the Status trait is making me wonder how the two would fit together. I have nothing to change here, but just thinking about it.